### PR TITLE
Add background task to fix potential SSO failure

### PR DESCRIPTION
**Pull request by jlawton**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jlawton has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m [hunk 1]
```diff
@@ -163,6 +167,40 @@ - (void)viewDidAppear:(BOOL)animated
...
+    [super viewWillDisappear:animated];
+}
+
+// Fixes a bug starting iOS 11.3, where backgrounding the app to respond to
+// 2FA could cause the login to fail.
+- (void)applicationWillResignActive:(NSNotification *)note
+{
+    UIApplication *app = [UIApplication sharedApplication];
+    [app beginBackgroundTaskWithExpirationHandler:^{
+        [app endBackgroundTask:self.backgroundTaskID];
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m:193]: weak self?

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m [hunk 2]
```diff
@@ -154,6 +156,8 @@ - (void)viewDidLoad
             me.completionBlock(me, user, error);
         }];
     }
+    
+    self.backgroundTaskID = UIBackgroundTaskInvalid;
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m]: any reason not to initialize this at init time?

**Review by boxtcarpel — APPROVED** [review]

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m (hunk 1 above)
Comment (jlawton) [reply BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m:193]: Shouldn't need it. There should be no way to deallocate self before this block is released.

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m (hunk 2 above)
Comment (jlawton) [reply BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m]: No. In fact, I think that Invalid == 0, and it's automatically initialized, but I'll move it.

**Pull request merged**